go/arrow/flight/client.go

```diff
@@ -348,10 +352,87 @@ func (c *client) Authenticate(ctx context.Context, opts ...grpc.CallOption) erro
 	return c.authHandler.Authenticate(ctx, &clientAuthConn{stream})
 }
 
+// Ensure the result of a DoAction is fully consumed
+func ReadUntilEOF(stream FlightService_DoActionClient) error {
```

Comment:
do we need this to be exposed and exported? If not, I'd prefer to keep it unexported unless you think it'll be specifically useful for users.